src/backend/cdb/dispatcher/cdbgang.c

```diff
@@ -808,6 +808,7 @@ GpResetSessionIfNeeded(void)
 
 		gp_session_id = newSessionId;
 		gp_command_count = 0;
+		MyProc->queryCommandId = 0;
```

Comment:
This patch changes the GpResetSessionIfNeeded function, but the original one changes CheckForResetSession in this file. Add please to the description why this change of the original patch is correct.